sycl/test/check_device_code/device_global_const_eval_use.cpp

```diff
@@ -0,0 +1,144 @@
+// RUN: %clangxx -std=c++20 -fsycl -c -fsycl-device-only -S -emit-llvm %s -o - | FileCheck %s
+
+// Tests that device_globals with device_image_scope property can be compile
+// time constant initialized and that the constant value appears in the IR
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental;
+
+// Constant int and array of ints device_globals
+constexpr device_global<int, decltype(properties(device_image_scope))> 
+   dg_int{5};
+constexpr device_global<int[3], decltype(properties(device_image_scope))>
+   dg_int_arr{5, 2, 3};
+
+// Constant char and array of char device_globals
+constexpr device_global<char, decltype(properties(device_image_scope))>
+    dg_char{'f'};
+constexpr device_global<char[3], decltype(properties(device_image_scope))>
+    dg_char_arr{'d', '4', 'S'};
+
+// Multidimensional array of integers
+constexpr device_global<int[3][2], decltype(properties(device_image_scope))>
+    dg_multi_dim_arr{3, 4, 5, 6, 7, 8};
+
+// Constant float and array of float device_globals
+constexpr device_global<float, decltype(properties(device_image_scope))>
+   dg_float{4.5};
+constexpr device_global<float[6], decltype(properties(device_image_scope))>
+   dg_float_arr{4.5, 2.1, 3.5, 9.33, 2.33, 2.1};
+   
+// Constant double and array of double device_globals
+constexpr device_global<double, decltype(properties(device_image_scope))>
+   dg_double{3.56543};
+constexpr device_global<double[3], decltype(properties(device_image_scope))>
+   dg_double_arr{2.2341234, 233.23423, 236.52321};
+
+// Constant bool and array of bool device_globals
+constexpr device_global<bool, decltype(properties(device_image_scope))>
+   dg_bool{true};
+constexpr device_global<bool[3], decltype(properties(device_image_scope))>
+   dg_bool_arr{true, false, true};
+
+// Constant struct and array of struct device_globals
+struct TestStruct {
+  int field1;
+  bool field2;
+  float field3;
+  int field4[4];
+};
+constexpr TestStruct TS1(5, true, 2.1, {1, 2, 3, 4});
+constexpr TestStruct TS2(7, false, 2.4, {1, 2, 3, 4});
+constexpr TestStruct TS3(6, false, 4.34534, {5, 6, 7, 8});
+constexpr device_global<TestStruct, decltype(properties(device_image_scope))>
+   dg_struct{TS3};
+constexpr device_global<TestStruct[2], decltype(properties(device_image_scope))>
+   dg_struct_arr{TS1, TS2};
+
+// Struct with constexpr constructor
+struct TestStruct2 {
+   int value;
+   constexpr TestStruct2(int val) : value(val) {}; 
+};
+constexpr TestStruct2 TS4{4};
+constexpr device_global<TestStruct2, decltype(properties(device_image_scope))>
+    dg_constexpr_constructor_struct{TS4};
+
+int main () {
+  sycl::queue Q;
+  Q.submit([&](sycl::handler& h) {
+    // Simple kernel that just copies over the values from the device_globals so
+    // that we can observe the GlobalVariables that are created to represent
+    // them in the IR
+    h.single_task([=] {
+      // Int and array of ints
+      int test_int = dg_int;
```

Comment:
```suggestion
      std::ignore = dg_int;
```

would make the test sources more readable. I'm not sure if the produced IR would become harder to reason about. Same everywhere else.

Reply:
Updated.